pkg/network/server.go

```diff
@@ -286,6 +286,32 @@ func (s *Server) handleInvCmd(p Peer, inv *payload.Inventory) error {
 	return p.WriteMsg(NewMessage(s.Net, CMDGetData, payload))
 }
 
+// handleInvCmd processes the received inventory.
+func (s *Server) handleGetDataCmd(p Peer, inv *payload.Inventory) error {
+	if !inv.Type.Valid() || len(inv.Hashes) == 0 {
```

Comment:
saw the same check in `handleInvCmd` so probably it will make sense to extract that as a separate func